### DISCUSSION TITLE
WARNING: Pygments lexer name 'ipython3' is not known

### DISCUSSION THREAD

**Discussion by tschm** [executablebooks/meta#890] [General]
This is a somewhat bizarre problem I have managed to run into recently. This particular bug has multiple years of history, though.
I can not produce it locally but I do get problems on our internal ci/cd engine. 

So here are a few lines to set the scene:

```
        - pip install jupyter-book pygments jupytext ipython
        - jupyter-book clean book
        - jupyter-book build book
```
This works as expected. However when running the same code on our ci/cd server I get tons of warnings and the code rendered results are poor:

`.../hydra/book/docs/notebooks/bson.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known`

I wonder whether there is a more explicit and slower process to identify the problem.

**Comment by choldgraf** [comment]
I believe this is an IPython bug that was introduced in IPython 8.7:

- https://github.com/ipython/ipython/issues/13845

The short term fix is to pin the version number, and the longer term fix is to wait for it to be fixed upstream and then update!

**Reply by tschm** [reply]
Thank you. Fixing the version did the job
